packages/web/src/pages/dashboard-page/components/AlbumsTab.tsx

```diff
@@ -0,0 +1,188 @@
+import { useMemo } from 'react'
+
+import { isContentUSDCPurchaseGated, Collection } from '@audius/common/models'
+import { accountSelectors } from '@audius/common/store'
+import { Nullable } from '@audius/common/utils'
+import {
+  IconVisibilityHidden,
+  IconVisibilityPublic,
+  IconCart,
+  Paper
+} from '@audius/harmony'
+import { useSelector } from 'react-redux'
+
+import { TracksTable, TracksTableColumn } from 'components/tracks-table'
+
+import { makeGetDashboard } from '../store/selectors'
+
+const { getAccountAlbums } = accountSelectors
+
+// Pagination Constants
+export const tablePageSize = 50
+
+const messages = {
+  public: 'Public',
+  premium: 'Premium',
+  hidden: 'Hidden'
+}
+
+const albumTableColumns: TracksTableColumn[] = [
+  'spacer',
+  'trackName',
+  'releaseDate',
+  'reposts',
+  'overflowMenu'
+]
+
+export type DataSourceAlbum = Collection & {
+  key: string
+  name: string
+  date: string
+  saves: number
+  reposts: number
+}
+
+export enum AlbumFilters {
+  PUBLIC = 'Public',
+  PREMIUM = 'Premium',
+  HIDDEN = 'Hidden'
+}
+
+const formatAlbumMetadata = (album: Collection): DataSourceAlbum => {
+  return {
+    ...album,
+    key: String(album.playlist_id),
+    name: album.playlist_name,
+    date: album.created_at,
+    saves: album.save_count,
+    reposts: album.repost_count
+  }
+}
+
+export const useArtistDashboardAlbumFilters = ({
```

Comment:
personally i'm a fan of co-location - since this hook is only meant to be used with this component, i like them in the same file. rather than a separate hooks file that has hooks related to two separate components. This makes it more clear that they go together. do u have strong feelings?

i like the refactor idea tho i'll give that a shot.

Reply:
nah you were right, too messy, separate files is good